common/cache/cache.go

```diff
@@ -102,6 +105,16 @@ type SimpleOptions struct {
 // deletion, Cache calls go f(i)
 type RemovedFunc func(interface{})
 
+// OnCacheUpdatedFunc is a type for notifying applications when the cache item was put/deleted
```

Comment:
nit: probably also update the comment as now the func will also be called when value is updated.